Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperationResult.swift

```diff
@@ -120,3 +149,141 @@ extension DeleteResult: BulkWriteResultConvertible {
+            } else if let invalidArgumentError = error as? InvalidArgumentError {
+                expect(invalidArgumentError.errorDescription.lowercased()).to(contain(errorContains))
+            } else if let connectionError = error as? ConnectionError {
+                expect(connectionError.message.lowercased()).to(contain(errorContains))
+            } else {
+                XCTFail("\(error) does not contain message")
+            }
+        }
+    }
+
+    // swiftlint:enable cyclomatic_complexity
+
+    internal func checkCodeName(_ error: Error) throws {
+        if let errorCodeName = self.errorCodeName {
+            if let commandError = error as? CommandError {
+                expect(commandError.codeName).to(satisfyAnyOf(equal(errorCodeName), equal("")))
+            } else if let writeError = error as? WriteError {
+                if let writeFailure = writeError.writeFailure {
+                    expect(writeFailure.codeName).to(satisfyAnyOf(equal(errorCodeName), equal("")))
+                }
+                if let writeConcernFailure = writeError.writeConcernFailure {
+                    expect(writeConcernFailure.codeName).to(satisfyAnyOf(equal(errorCodeName), equal("")))
+                }
+            } else if let bulkWriteError = error as? BulkWriteError {
+                if let writeFailures = bulkWriteError.writeFailures {
+                    for writeFailure in writeFailures {
+                        expect(writeFailure.codeName).to(satisfyAnyOf(equal(errorCodeName), equal("")))
+                    }
+                }
+                if let writeConcernFailure = bulkWriteError.writeConcernFailure {
+                    expect(writeConcernFailure.codeName).to(satisfyAnyOf(equal(errorCodeName), equal("")))
+                }
+            } else {
+                XCTFail("\(error) does not contain codeName")
+            }
+        }
+    }
+
+    internal func checkErrorLabels(_ error: Error) throws {
+        // `configureFailPoint` command correctly handles error labels in MongoDB v4.3.1+ (see SERVER-43941).
+        // Do not check the "RetryableWriteError" error label until the spec test requirements are updated.
+        let skippedErrorLabels = ["RetryableWriteError"]
+
+        if let errorLabelsContain = self.errorLabelsContain {
+            guard let labeledError = error as? LabeledError else {
+                XCTFail("\(error) does not contain errorLabels")
+                return
+            }
+            for label in errorLabelsContain where !skippedErrorLabels.contains(label) {
+                expect(labeledError.errorLabels).to(contain(label))
+            }
+        }
+
+        if let errorLabelsOmit = self.errorLabelsOmit {
+            guard let labeledError = error as? LabeledError else {
+                XCTFail("\(error) does not contain errorLabels")
+                return
+            }
+            if labeledError.errorLabels == nil {
```

Comment:
it would be a bit more idiomatic to write this as a `guard let`. I think this could alternatively be omitted as it is in the errorLabelsContain part.

Reply:
I included this here because an error could conform to `LabeledError` but have `errorLabels=nil`, in which case the expectation fails.

Reply:
I've changed it to a `guard let`.

Reply:
Should we do the same thing for the errorLabelsContain portion then?

Reply:
I don't think so because we want the `expect` to fail in errorLabelsContain if `labeledError.errorLabels` is `nil`.

Reply:
Oh right, of course. I got myself confused.